app/controllers/admin/products_v3_controller.rb

```diff
@@ -87,14 +87,16 @@ def ransack_query
 
     # Optimise by pre-loading required columns
     def product_query_includes
-      # TODO: add other fields used in columns? (eg supplier: [:name])
       [
-        # variants: [
-        #   :default_price,
-        #   :stock_locations,
-        #   :stock_items,
-        #   :variant_overrides
-        # ]
+        :image,
+        :supplier,
+        { variants: [
+          :default_price,
+          :primary_taxon,
+          :product,
```

Comment:
Do you need the product here ? isn't the request to load product ?

Reply:
Good question. I think I added it because it was suggested by the Bullet gem. But surely Rails is smart enough to avoid needing this, let me see...

Reply:
I don't know why, but removing this generates 14 additional queries in my test. (Lucky I didn't squash it so it was easy to test again 😉 )
Rails should be [finding inverse associations automatically](https://github.com/rails/rails/pull/9522#top), and even if I explicitly define `inverse_of`, it still doesn't help:

```
    belongs_to :product, -> { with_deleted }, touch: true, class_name: 'Spree::Product', inverse_of: :variants
```

Part of me wants to dig further to find out the answer, but I'm sure it's not worth it!

Reply:
Thanks for looking into it, I am wondering if that could come from calling `variant.product` somewhere in the code. You'd think we shouldn't need that as we are dealing with products but who knows ? :man_shrugging: